components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3692,5 +3249,628 @@ subroutine get_time_space_phys_variables( &
 
 end subroutine get_time_space_phys_variables
 
+subroutine cloud_sedimentation(kts,kte,    &
+   qc_incld,rho,inv_rho,lcldm,acn,inv_dzq,&
+   dt,odt,dnu,lammin,lammax,log_predictNc, & 
+   qc, nc, nc_incld,mu_c,nu,lamc,prt_liq,qc_tend,nc_tend) 
+   
+   implicit none 
+   integer, intent(in) :: kts, kte
+
+   real(rtype), intent(in), dimension(kts:kte) :: qc_incld
+
+   real(rtype), intent(in), dimension(kts:kte) :: rho
+   real(rtype), intent(in), dimension(kts:kte) :: inv_rho 
+   real(rtype), intent(in), dimension(kts:kte) :: lcldm 
+   real(rtype), intent(in), dimension(kts:kte) :: acn 
+   real(rtype), intent(in), dimension(kts:kte) :: inv_dzq
```

Comment:
its worth noting that we had to make some adjustments to the original P3 sedimentation code with regards to dzq.  I never considered if that would impact how dzq is used in places other than the movement of mass down the column.  I want to note here that we may want to check how the definition of dzq impacts `Co_max` and if we need to make any changes here.